wordpress-mobile/WordPress-iOS
# Blaze Manage Campaigns: Present campaign details view from campaigns list

Part of #20765 

## Description
This PR adds an API to `BlazeFlowCoordinator` for presenting the details web view for a campaign. The existing Blaze web views logic was refactored and reused. The PR also hooks up the new API to the campaigns list.

The following are still missing:
- Analytics tracking
- Presenting the details view from the dashboard card
- Use remotely configured params for the web view URLs instead of constants (Ref: #20766)

## Screenshots

<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/25306722/cb4a228e-c3e5-42d1-9570-c33fcd9a9767" width=40%>

## Testing Instructions

Before testing, replace `BlazeCampaignsViewController.fetchCampaigns()` with the following:
```swift
private func fetchCampaigns() {
    isLoading = true

    BlazeService.createService()?.getRecentCampaigns(for: blog, completion: { [weak self] (result) in
        self?.isLoading = false
        switch result {
        case .success(let success):
            self?.campaigns = success.campaigns ?? []
        case .failure(let failure):
            print("Failed to fetch recent blaze campaigns: \(failure.localizedDescription)")
            break
        }
    })
}
```

1. Run the Jetpack app
2. Log in using the test account mentioned here: p1687419588093579/1687363386.825549-slack-C04LJFS1G5P 
3. Tap on the Blaze dashboard card header
4. The campaigns list should be displayed and populated
5. Tap on any campaign
6. Make sure the campaign details are presented in a web view ✅ 

## Regression Notes
1. Potential unintended areas of impact
The original blaze flow for creating a campaign.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Relied partially on `BlazeCreateCampaignWebViewModelTests` but also tested manually.

3. What automated tests I added (or what prevented me from doing so)
`BlazeCampaignDetailsWebViewModelTests`

PR submission checklist:


UI Changes testing checklist:
